core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -142,72 +283,150 @@ public TermsQueryBuilder disableCoord(boolean disableCoord) {
         return this;
     }
 
+    public boolean disableCoord() {
+        return this.disableCoord;
+    }
+
+    public boolean isTermsLookupQuery() {
+        return this.termsLookup != null;
+    }
+
+    public TermsQueryBuilder termsLookup(TermsLookupBuilder termsLookup) {
+        this.termsLookup = termsLookup;
+        return this;
+    }
+
+    public TermsLookupBuilder termsLookup() {
+        return this.termsLookup;
+    }
+
     /**
      * Sets the index name to lookup the terms from.
      */
     public TermsQueryBuilder lookupIndex(String lookupIndex) {
-        this.lookupIndex = lookupIndex;
+        if (lookupIndex != null) {
```

Comment:
I am fine either way as long as we settle on something. If you ask me I would already have removed all setters, TermsLookupQueryBuilder, and TermsLookup and just keep TermsQueryBuilder with TermsLookupBuilder. There would then be only one way of doing terms lookup queries and/or builder terms lookup objects.